### ISSUE TITLE
regtest doesn't work

### ISSUE THREAD

**Issue by lilione** [emercoin/emercoin#72]
I want to write some functional tests. But the test framework in `rpc-tests` seems copied from bitcoin without any modification. So I made some modification by myself. The problem came when I want to call `generate()` to generate new blocks. It failed as the blocks generated couldn't pass the difficulty check. Then I noticed that the nBits for genesis block for regtest is only `0x1d0fffff` compared to `0x207fffff` in bitcoin, which makes it hard to generate new blocks. So I tried to modify the nBits for regtest. Then in `kernel.cpp`, function `GetStakeModifierChecksum` failed at `assert (pindex->pprev || pindex->GetBlockHash() == Params().GetConsensus().hashGenesisBlock);`. The `pindex->nBits` is still `0x1d0fffff` even though I changed nBits for `CreateGenesisBlock`. Can anyone tell me how to modify the difficulty that makes regtest work?

**Comment by EvgenijM86** [comment]
Yes. I didn't implement regtest yet, because no one have asked for it before.

I think you need to recreate genesis block with new nBits. There is a MineGenesisBlock() function in src/chainparams.cpp that can do just that.

**Comment by EvgenijM86** [comment]
As for difficulty - you can try changing parameters in class CRegTestParams : public CChainParams. Like: consensus.powLimit and consensus.bnInitialHashTarget. I think you need to increase their value in order to make it easier to mine.

**Comment by maxihatop** [comment]
No, we do not support it.
